### PR TITLE
Pulsar Client: allow to print GenericRecord contents

### PR DESCRIPTION
**Pull request by eolivelli**
### Motivation
When you are dealing with structured data it would be pretty useful to have a tool that prints decoded records directly to the console.

### Modifications

Add two new options to the "pulsar-client consume" tool:
- "--schema-type" option, that can be "bytes" and "auto_consume" in order to select the Schema (in particular you are interested in AUTO_CONSUME if you are dealing with GenericRecords)
- "--hide-content" boolean option, that shuts down the printing of the contents of the messages, in case they are in binary form or that you are using the tool only to test if any message is coming from the topic

In case of GenericRecords we are printing the content of the record like a Java Map. We are also handling nested GenericRecord structures as well.

### CODE REVIEW

**Comment by eolivelli** [comment]
@codelipenghui @wolfstudy @rdhabalia please help merging this patch.
it is a trivial work but very useful while dealing with structured data

**Review by codelipenghui — APPROVED** [review]

File: pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java [hunk 1]
```diff
@@ -31,30 +31,19 @@
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Base64;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (eolivelli) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java]: @codelipenghui  it was my IDE.

we should have checkstyle to block this issue.

[the remark this answers is not in the document]
Comment (eolivelli) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java]: I have fixed it

[the remark this answers is not in the document]
Comment (codelipenghui) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java]: Yes, I think the check style plugin does not enable for this module, so...

**Comment by Anonymitaet** [comment]
@eolivelli thanks for your work, could you pls add docs for this PR?

**Review by zymap — APPROVED** [review]

**Comment by eolivelli** [comment]
@Anonymitaet I added docs in the pulsar-client section.
Thanks for the suggestion
PTAL

**Review by Anonymitaet — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@eolivelli Many thanks for your docs.

**Comment by eolivelli** [comment]
@codelipenghui  @zymap this patch has been approved, can you please help me merging to master branch ?
thanks for your review (also to @Anonymitaet )

**Review by 315157973 — APPROVED** [review]

**Comment by eolivelli** [comment]
@codelipenghui @rdhabalia  I am not sure why pulsarbot it not going to rerun the failed tests.
can you please force the rerun from the github UI ?

**Comment by codelipenghui** [comment]
@eolivelli I have rerun through the Github UI.

**Comment by eolivelli** [comment]
Tests are always failing for errors unrelated to this patch.
I am merging with current master, let's see if it goes better

**Pull request merged**